### ISSUE TITLE
[BUG] GroupKFold

### ISSUE THREAD

**Issue by rspadim** [pycaret/pycaret#1624]
I was trying groupkfold split and i had errors, i think kfoldgroup values aren't being splitted at internal functions
what i did:

```
df_ml = <some machine learning data set, where we have 1 column as datetime, and this column have +- 3 different dates>
datas_map = {l: i for i, l in enumerate(df_ml['datetime_column'].dt.date.unique())}
df_ml['fold_groups'] = (df_ml['datetime_column'].dt.date).map(datas_map)

new_exp = pycaret.classification.setup(
        data=df_ml[allowed_features],
        target='target',
        session_id=1,
        fold=min(10, len(datas_map)),
        fold_strategy='groupkfold',
        fold_groups=df_ml['fold_groups'],
        numeric_features=['some numeric feature classified as categorical'],
    )


best_model = pycaret.classification.compare_models(
    #groups=df_ml['fold_groups']   #<- here, this doen't work, but it's not the issue bug
)

model = pycaret.classification.create_model(
    'lightgbm',
    #groups=df_ml['fold_groups']   #<- here, this doen't work, but it's not the issue bug
)

--- here at create_model i got an exception
```

exception:
ValueError: groups has lenght 1624 which doesn't match X_train length of 1136.

i think it's a problem with split function internally, maybe something like:
receive the dataframe, train test split (70%), train the model but forgetting the train_test_split of kfoldgroup data 

---
create model
Traceback (most recent call last):
  File "test.py", line 144, in <module>
    model = pycaret.classification.create_model(
  File "C:\ProgramData\Anaconda3\lib\site-packages\pycaret\classification.py", line 890, in create_model
    return pycaret.internal.tabular.create_model_supervised(
  File "C:\ProgramData\Anaconda3\lib\site-packages\pycaret\internal\tabular.py", line 2969, in create_model_supervised
    groups = _get_groups(groups, data=X_train_data)
  File "C:\ProgramData\Anaconda3\lib\site-packages\pycaret\internal\tabular.py", line 10483, in _get_groups
    return pycaret.internal.utils.get_groups(groups, data, fold_groups)
  File "C:\ProgramData\Anaconda3\lib\site-packages\pycaret\internal\utils.py", line 557, in get_groups
    raise ValueError(
ValueError: groups has lenght 1624 which doesn't match X_train length of 1136.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Program Files\JetBrains\PyCharm 2020.3.3\plugins\python\helpers\pydev\pydevd.py", line 1477, in _exec
    pydev_imports.execfile(file, globals, locals)  # execute the script
  File "C:\Program Files\JetBrains\PyCharm 2020.3.3\plugins\python\helpers\pydev\_pydev_imps\_pydev_execfile.py", line 18, in execfile
    exec(compile(contents+"\n", file, 'exec'), glob, loc)
  File "test.py", line 150, in <module>
    model = pycaret.classification.create_model(
  File "C:\ProgramData\Anaconda3\lib\site-packages\pycaret\classification.py", line 890, in create_model
    return pycaret.internal.tabular.create_model_supervised(
  File "C:\ProgramData\Anaconda3\lib\site-packages\pycaret\internal\tabular.py", line 2969, in create_model_supervised
    groups = _get_groups(groups, data=X_train_data)
  File "C:\ProgramData\Anaconda3\lib\site-packages\pycaret\internal\tabular.py", line 10483, in _get_groups
    return pycaret.internal.utils.get_groups(groups, data, fold_groups)
  File "C:\ProgramData\Anaconda3\lib\site-packages\pycaret\internal\utils.py", line 557, in get_groups
    raise ValueError(
ValueError: groups has lenght 1624 which doesn't match X_train length of 1136.

Process finished with exit code 1

**Comment by ekamperi** [comment]
Hey folks! Is there any update on this one? I just encountered it. What if I use `test_data=training_data` in `setup()`? Would that be a valid workaround? `compare_models()` & `tune_model()` would work ok, right? But it wouldn't make sense to call `predict_model()` then, unless I stash, say, 30% of the data *before* passing it to `setup()`. Correct?

**Comment by RxstydnR** [comment]
I just encountered the same problem when I use group kfold on timeseries dataset. The timeseries dataset has **multiple duplicated index**, and that caused the problem, in my situation. I am not sure this is related to your problem, but I hope it helps you.

**Comment by tvdboom** [comment]
This issue has been fixed in #3057
